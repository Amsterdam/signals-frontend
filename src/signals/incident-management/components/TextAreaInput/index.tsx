// SPDX-License-Identifier: MPL-2.0
// Copyright (C) 2018 - 2022 Gemeente Amsterdam
import type { ChangeEvent } from 'react'

import styled from 'styled-components'

import Label from 'components/Label'
import TextArea from 'components/TextArea'

const Wrapper = styled.div`
  width: 100%;
  margin-bottom: 20px;
`

type Props = {
  name: string
  value: string
  display: string
  placeholder: string
  onChange: (e: ChangeEvent<HTMLTextAreaElement>) => void
  maxLength?: number
  rows?: number
}

const TextAreaInput = (props: Props) => {
  const { name, value, display, placeholder, rows, maxLength, onChange } = props
  return (
    <Wrapper>
      <Label htmlFor={`form${name}`}>{display}</Label>

      <TextArea
        id={`form${name}`}
        name={name}
        data-testid={name}
        value={value}
        onChange={onChange}
        placeholder={placeholder}
        rows={rows}
        infoText={
          maxLength &&
          maxLength > 0 &&
          `${value ? value.length : '0'}/${maxLength} tekens`
        }
      />
    </Wrapper>
  )
}

export default TextAreaInput
