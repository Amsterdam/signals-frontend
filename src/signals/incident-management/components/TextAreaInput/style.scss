.text-area-input {
  width: 100%;

  &__label {
    margin-bottom: 4px;
  }

  &__control {
    margin-bottom: 16px;
  }
}
