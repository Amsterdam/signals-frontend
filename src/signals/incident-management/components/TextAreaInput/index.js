import React from 'react';
import PropTypes from 'prop-types';

import './style.scss';

const TextAreaInput = (props) => {
  const { name, display, placeholder, rows, maxLength } = props;
  const render = ({ handler, value }) => (
    <div className="text-area-input">
      <div className="mode_input text rij_verplicht">
        <div className="text-area-input__label">
          <label htmlFor={`form${name}`}>{display}</label>
        </div>

        <div className="text-area-input__control invoer">
          <textarea
            name=""
            id={`form${name}`}
            value=""
            {...handler()}
            placeholder={placeholder}
            rows={rows}
          />
          { maxLength &&
            <div className="input-help">
              <span className="text-area-input__counter">
                {`${value ? value.length : '0'}/${maxLength} tekens` }
              </span>
            </div>
          }
        </div>
      </div>
    </div>);

  render.defaultProps = {
    touched: false,
    placeholder: '',
    rows: 4
  };

  render.propTypes = {
    handler: PropTypes.func.isRequired,
    value: PropTypes.string,
    maxLength: PropTypes.number
  };
  return render;
};

export default TextAreaInput;
