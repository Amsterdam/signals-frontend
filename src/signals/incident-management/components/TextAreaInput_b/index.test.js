// SPDX-License-Identifier: MPL-2.0
// Copyright (C) 2018 - 2021 Gemeente Amsterdam
import Enzyme, { shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

import TextAreaInput from '.'

Enzyme.configure({ adapter: new Adapter() })

describe('<TextAreaInput />', () => {
  let wrapper
  let props

  beforeEach(() => {
    props = {
      name: 'name',
      display: 'display',
      placeholder: 'placeholder',
      handler: jest.fn(),
      rows: 2,
    }

    const TextAreaInputRender = TextAreaInput(props)
    wrapper = shallow(<TextAreaInputRender {...props} />)
  })

  afterEach(() => {
    jest.resetAllMocks()
  })

  it('should render correctly', () => {
    const TextAreaInputRender = TextAreaInput(props)
    wrapper = shallow(<TextAreaInputRender {...props} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should render character counter correctly', () => {
    props.maxLength = 300
    const TextAreaInputRender = TextAreaInput(props)
    wrapper = shallow(<TextAreaInputRender {...props} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should render character counter with value correctly', () => {
    props.maxLength = 300
    props.value = 'test'
    const TextAreaInputRender = TextAreaInput(props)
    wrapper = shallow(<TextAreaInputRender {...props} />)

    expect(wrapper).toMatchSnapshot()
  })
})
