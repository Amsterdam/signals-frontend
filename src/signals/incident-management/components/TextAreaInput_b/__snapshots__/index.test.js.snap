// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextAreaInput /> should render character counter correctly 1`] = `
<styled.div>
  <styled.label
    htmlFor="formname"
  >
    display
  </styled.label>
  <ForwardRef
    data-testid="name"
    id="formname"
    infoText="0/300 tekens"
    name="name"
    placeholder="placeholder"
    rows={2}
  />
</styled.div>
`;

exports[`<TextAreaInput /> should render character counter with value correctly 1`] = `
<styled.div>
  <styled.label
    htmlFor="formname"
  >
    display
  </styled.label>
  <ForwardRef
    data-testid="name"
    id="formname"
    infoText="4/300 tekens"
    name="name"
    placeholder="placeholder"
    rows={2}
    value="test"
  />
</styled.div>
`;

exports[`<TextAreaInput /> should render correctly 1`] = `
<styled.div>
  <styled.label
    htmlFor="formname"
  >
    display
  </styled.label>
  <ForwardRef
    data-testid="name"
    id="formname"
    name="name"
    placeholder="placeholder"
    rows={2}
  />
</styled.div>
`;
